packages/dev-utils/src/subproviders/subprovider.ts

```diff
@@ -0,0 +1,69 @@
+import { promisify } from '@0x/utils';
+import { JSONRPCRequestPayload, JSONRPCResponsePayload } from 'ethereum-types';
+import Web3ProviderEngine = require('web3-provider-engine');
+
+import { Callback, ErrorCallback } from './types';
+/**
+ * Use the subprovider in @0x/subproviders. This is internal for dev-utils only.
+ * Using any below avoiding types due to cyclic dependencies
+ */
+export abstract class Subprovider {
+    // tslint:disable-next-line:underscore-private-and-protected
+    private engine!: Web3ProviderEngine;
+    protected static _createFinalPayload(payload: Partial<JSONRPCRequestPayload>): Partial<JSONRPCRequestPayload> {
+        const finalPayload = {
+            // defaults
+            id: Subprovider._getRandomId(),
+            jsonrpc: '2.0',
+            params: [],
```

Comment:
wouldn't you want to use the params that came in from `payload` here?

Reply:
It gets overwritten on line 19 with the exploded `...payload`, but this syntax has a default of [] if not provided.